### PR TITLE
feat: add support for OpenTofu plan files

### PR DESCRIPTION
**Pull request by snakster**
## What this PR does / why we need it:
This PR allows syncing plan files that were created with `tofu`to Terramate cloud.
A new option `--tofu-plan-file`/`tofu_plan_file` is added that can be used instead of `--terraform-plan-file`/`terraform_plan_file`.

When the tofu options are used, the following changes:
- The provisioner field of the plan data sent to be backend is set to `opentofu`.
- The internal invocation to get the plan details uses `tofu show` instead of `terraform show`.

## Does this PR introduce a user-facing change?

```
Yes, it adds a flag and a command option.
```

### CODE REVIEW

File: cmd/terramate/e2etests/cloud/run_cloud_drift_test.go [hunk 1]
```diff
@@ -683,7 +761,12 @@ func assertRunDrifts(t *testing.T, cloudData *cloudstore.Data, tmcAddr string, e
 		assert.NoError(t, json.Unmarshal([]byte(got.Details.ChangesetJSON), &gotPlan))
 		assert.NoError(t, json.Unmarshal([]byte(expected.Details.ChangesetJSON), &wantPlan))
 
-		if diff := cmp.Diff(gotPlan, wantPlan, cmpopts.IgnoreFields(tfjson.Plan{}, "Timestamp", "FormatVersion")); diff != "" {
+		if diff := cmp.Diff(gotPlan, wantPlan,
+			// TODO (snk): Could also supply two planfiles, but I think this is fine, too.
```
Comment (i4ki) [inline cmd/terramate/e2etests/cloud/run_cloud_drift_test.go:765]: 👍

**Review by i4ki — APPROVED** [review]
LGTM

The `ci.yml` comment is minor, we can remove later. As our CI is already very slow, better not make it slower for no reason.

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -84,10 +90,17 @@ jobs:
...
       - uses: actions/setup-go@v5
         with:
           go-version: '1.21'
 
+      - uses: opentofu/setup-opentofu@v1
+        with:
+          tofu_version: 1.6.2
+          tofu_wrapper: false
+          github_token: ${{ secrets.GITHUB_TOKEN }}
+
```
[the remark this answers is not in the document]
Comment (snakster) [reply .github/workflows/ci.yml]: Thanks! Fixed

**Review by i4ki — APPROVED** [review]
lgtm

**Pull request merged**